apps/entity-service/src/entities/dto/site-report.dto.ts

```diff
@@ -0,0 +1,302 @@
+    description: "The associated site uuid"
+  })
+  siteUuid: string | null;
+
+  @ApiProperty()
+  frameworkKey: string | null;
+
+  @ApiProperty()
+  frameworkUuid: string | null;
+
+  @ApiProperty()
+  status: string;
+
+  @ApiProperty()
+  updateRequestStatus: string;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated project name"
+  })
+  projectName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated project uuid"
+  })
+  projectUuid: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation name"
+  })
+  organisationName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation uuid"
+  })
+  organisationUuid: string | null;
+
+  @ApiProperty()
+  updatedAt: Date;
+
+  @ApiProperty({ nullable: true })
+  submittedAt: Date | null;
+
+  @ApiProperty({ nullable: true })
+  taskId: number | null;
+
+  @ApiProperty()
+  dueAt: Date | null;
+
+  @ApiProperty({ nullable: true })
+  reportTitle: string | null;
+
+  @ApiProperty()
+  createdAt: Date;
+}
+
+export type AdditionalSiteReportLightProps = Pick<SiteReportLightDto, "reportTitle">;
+export type AdditionalSiteReportFullProps = AdditionalSiteReportLightProps &
+  AdditionalProps<SiteReportFullDto, SiteReportLightDto & Omit<SiteReport, "site">>;
+export type SiteReportMedia = Pick<SiteReportFullDto, keyof typeof SiteReport.MEDIA>;
+
+export class SiteReportFullDto extends SiteReportLightDto {
+  constructor(siteReport: SiteReport, props?: AdditionalSiteReportFullProps) {
+    super();
+    this.populate(SiteReportFullDto, {
+      ...pickApiProperties(siteReport, SiteReportFullDto),
+      lightResource: false,
+      // these two are untyped and marked optional in the base model.
+      createdAt: siteReport.createdAt as Date,
+      updatedAt: siteReport.createdAt as Date,
+      ...props
+    });
+  }
+
+  @ApiProperty({ nullable: true })
+  reportTitle: string | null;
+
+  @ApiProperty({ nullable: true })
+  projectReportTitle: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated site name"
+  })
+  siteName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated site uuid"
+  })
+  siteUuid: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation name"
+  })
+  organisationName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation uuid"
+  })
+  organisationUuid: string | null;
+
+  @ApiProperty()
+  dueAt: Date | null;
+
+  @ApiProperty()
+  status: string;
+
+  @ApiProperty()
+  updateRequestStatus: string;
+
+  @ApiProperty({ nullable: true })
+  feedback: string | null;
+
+  @ApiProperty({ nullable: true })
+  feedbackFields: string[] | null;
+
+  @ApiProperty()
+  nothingToReport: boolean;
+
+  @ApiProperty()
+  readableCompletionStatus: string;
```

Comment:
The FE has everything it needs to generate a readable string from `status`; please don't calculate this in the BE.